app/Components/PlayerController/PlayerController.tsx

```diff
@@ -0,0 +1,239 @@
+
+import { useEffect, useRef, useState } from "react";
+import styles from "./PlayerController.module.scss";
+import songs from "@/public/Consts/songs"; 
+import Icon from "../Icon/Icon";
+
+type PlayerControllerProps = {
+  albumTitle: string;
+  dropdown: string;
+  queueTrack: string;
+  queueArtist: string;
+  photo: string;
+  image: string;
+  duration: number;
+  isPlaying: boolean;
+  onPlayPause: () => void;
+  onSkipForward: () => void;
+  onSkipBackward: () => void;
+  onRepeat: () => void;
+  onShuffle: () => void;
+  onToggleView: () => void; 
+  currentSongIndex: number; 
+  setCurrentSongIndex: React.Dispatch<React.SetStateAction<number>>; 
+};
+
+const PlayerController = (props: PlayerControllerProps) => {
```

Comment:
also this audioRef should be as helper for other Modes

Reply:
i will give audioRef as prop